E0512 needs to be updated to new format

From: src/test/compile-fail/E0512.rs

E0512 needs a span_label, updating it from:

```
error[E0512]: transmute called with differently sized types: u16 (16 bits) to u8 (8 bits)
  --> src/test/compile-fail/E0512.rs:14:23
   |
14 |     unsafe { takes_u8(::std::mem::transmute(0u16)); } //~ ERROR E0512
   |                       ^^^^^^^^^^^^^^^^^^^^^
```

to:

```
error[E0512]: transmute called with differently sized types: u16 (16 bits) to u8 (8 bits)
  --> src/test/compile-fail/E0512.rs:14:23
   |
14 |     unsafe { takes_u8(::std::mem::transmute(0u16)); } //~ ERROR E0512
   |                       ^^^^^^^^^^^^^^^^^^^^^ transmuting between 16 bits and 8 bits
```

--- a/src/librustc/middle/intrinsicck.rs
+++ b/src/librustc/middle/intrinsicck.rs
@@ -103,11 +103,16 @@ impl<'a, 'gcx, 'tcx> ExprVisitor<'a, 'gcx, 'tcx> {
             }
         };
 
-        span_err!(self.infcx.tcx.sess, span, E0512,
+        struct_span_err!(self.infcx.tcx.sess, span, E0512,
                   "transmute called with differently sized types: \
                    {} ({}) to {} ({})",
                   from, skeleton_string(from, sk_from),
-                  to, skeleton_string(to, sk_to));
+                  to, skeleton_string(to, sk_to))
+            .span_label(span,
+                &format!("transmuting between {} and {}",
+                    skeleton_string(from, sk_from),
+                    skeleton_string(to, sk_to)))
+            .emit();
     }
 }
 

--- a/src/test/compile-fail/E0512.rs
+++ b/src/test/compile-fail/E0512.rs
@@ -12,4 +12,5 @@ fn takes_u8(_: u8) {}
 
 fn main() {
     unsafe { takes_u8(::std::mem::transmute(0u16)); } //~ ERROR E0512
+                                                      //~| transmuting between 16 bits and 8 bits
 }

